Library/ViewModels/LiveStreamEventDetailsViewModel.swift

```diff
@@ -0,0 +1,267 @@
+
+  //swiftlint:disable:next function_body_length
+  public init () {
+    let configData = Signal.combineLatest(
+      self.configData.signal.skipNil(),
+      self.viewDidLoadProperty.signal
+      )
+      .map(first)
+
+    let project = configData.map(first)
+
+    let eventEvent = configData
+      .switchMap { project, liveStream, optionalEvent in
+        fetchEvent(forProject: project, liveStream: liveStream, event: optionalEvent)
+          .materialize()
+    }
+
+    let event = eventEvent.values()
+
+    self.retrievedLiveStreamEvent = event
+
+    self.configureShareViewModel = Signal.combineLatest(project, event)
+
+    self.shareButtonEnabled = self.configureShareViewModel.mapConst(true)
+
+    let subscribedProperty = MutableProperty(false)
+
+    let subscribeTappedOnLogin = Signal.combineLatest(
+      self.subscribeButtonTappedProperty.signal,
+      self.userSessionStartedProperty.signal
+      )
+      .take(first: 1)
+
+    let subscribeIntent = Signal.merge(
+      subscribeTappedOnLogin.ignoreValues(),
+      self.subscribeButtonTappedProperty.signal.filter { AppEnvironment.current.currentUser != nil }
+    )
+
+    let isSubscribedEvent = event
+      .takeWhen(subscribeIntent)
+      .switchMap { event -> SignalProducer<Event<Bool, LiveApiError>, NoError> in
+        guard let userId = AppEnvironment.current.currentUser?.id else { return .empty }
+
+        return AppEnvironment.current.liveStreamService.subscribeTo(
+          eventId: event.id, uid: userId, isSubscribed: subscribedProperty.value
+          )
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+    }
+
+    subscribedProperty <~ Signal.merge(
+      // Bind the API response values for subscribed
+      isSubscribedEvent.values(),
+
+      // Bind the initial subscribed value
+      event.map { $0.user.isSubscribed }
+    )
+
+    let subscribed = subscribedProperty.signal
+
+    self.availableForText = event
+      .map { event -> String? in
+        guard let availableDate = AppEnvironment.current.calendar
+          .date(byAdding: .day, value: 2, to: event.stream.startDate)?.timeIntervalSince1970
+          else { return nil }
+
+        let (time, units) = Format.duration(secondsInUTC: availableDate, abbreviate: false)
+
+        return Strings.Available_to_watch_for_time_more_units(time: time, units: units)
+      }.skipNil()
+
+    self.creatorAvatarUrl = event
+      .map { URL(string: $0.creator.avatar) }
+
+    self.openLoginToutViewController = self.subscribeButtonTappedProperty.signal
+      .filter { AppEnvironment.current.currentUser == nil }
+
+    self.creatorName = event.map { $0.creator.name }
+    self.liveStreamTitle = event.map { $0.stream.name }
+    self.liveStreamParagraph = event.map { $0.stream.description }
+
+    self.subscribeButtonImage = subscribed.map {
+      $0 ? "postcard-checkmark" : nil
+    }
```

Comment:
Unfortunately this doesn't work with the way that I'm using the subscribe button's image - I'm placing a checkmark to the right of the text using `self.subscribeButton.semanticContentAttribute = .forceRightToLeft`. Setting the button's `imageView.isHidden` to `true` doesn't seem to work and I also think if it did work it may not clear the space properly.